### PR TITLE
Extract "open-editor" utility

### PR DESCRIPTION
**Pull request by Turbo87**
This PR extracts an "open-editor" utility from the "edit-file-diff" code and adds tests for it.

### CODE REVIEW

File: lib/utilities/open-editor.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var Promise = require('../ext/promise');
+var spawn = require('child_process').spawn;
+
+var openEditor = function(file) {
```
Comment (stefanpenner) [inline lib/utilities/open-editor.js]: Typically do function foo() form

**Comment by stefanpenner** [comment]
Looks good

File: lib/utilities/open-editor.js (hunk 1 above)
Comment (Turbo87) [inline lib/utilities/open-editor.js]: changed

**Comment by Turbo87** [comment]
@homu r+

**Comment by Turbo87** [comment]
@homu retry

**Pull request merged**